app/com/yetu/oauth2provider/oauth2/models/YetuUser.scala

```diff
@@ -3,74 +3,83 @@ package oauth2
 package models
 
 import com.yetu.oauth2provider.signature.models.YetuPublicKey
+import com.yetu.oauth2provider.utils.DateUtility
 import com.yetu.oauth2resource.model.ContactInfo
-import play.api.libs.json.{ Format, Json }
+import org.joda.time.DateTime
+import play.api.libs.json.Json
 import securesocial.controllers.UserAgreement
-import securesocial.core._
-import _root_.java.util.Date
-
-import securesocial.core.{ OAuth2Info, OAuth1Info }
-
-//TODO: deprecated, get rid of this.
-case class IdentityId(userId: String, providerId: String)
+import securesocial.core.{ OAuth1Info, OAuth2Info, _ }
 
 /**
  * An implementation of user Profile.  Used to gather user information when users sign up and/or sign in.
  */
-case class YetuUser(identityId: IdentityId,
-    uid: String,
-    firstName: String,
-    lastName: String,
-    fullName: String,
-    email: Option[String],
-    avatarUrl: Option[String],
-    authMethod: AuthenticationMethod,
-    oAuth1Info: Option[OAuth1Info] = None,
-    oAuth2Info: Option[OAuth2Info] = None,
-    passwordInfo: Option[PasswordInfo] = None,
-    registrationDate: Option[Date] = None,
-    contactInfo: Option[ContactInfo] = None,
-    imageUrl: Option[String] = None,
-    publicKey: Option[YetuPublicKey] = None,
-    userAgreement: Option[UserAgreement] = None) {
-
-  //TODO: clean this up.
-  val userId = identityId.userId
-  val providerId = identityId.providerId
-  val main: BasicProfile = BasicProfile(
-    providerId,
-    userId,
-    Some(firstName),
-    Some(lastName),
-    Some(fullName),
-    email,
-    avatarUrl,
-    authMethod,
-    oAuth1Info,
-    oAuth2Info,
-    passwordInfo,
-    userAgreement)
-
-  val identities: List[BasicProfile] = List(main)
-
-  def toBasicProfile = BasicProfile(
-    providerId,
-    userId,
-    Some(firstName),
-    Some(lastName),
-    Some(fullName),
-    email,
-    avatarUrl,
-    authMethod,
-    oAuth1Info,
-    oAuth2Info,
-    passwordInfo,
-    userAgreement)
+class YetuUser(
```

Comment:
With all the stuff in the `copy` method below... isn't this a good candidate for a `case class`?